src/Commands/Remote/SSHBaseCommand.php

```diff
@@ -120,6 +124,10 @@ protected function validateEnvironment($site, $environment)
      */
     protected function validateConnectionMode($mode)
     {
+        // Skip the warning if it is disabled
+        if ($this->getConfig()->get('hide_git_mode_warning')) {
```

Comment:
I think this logic ought to apply directly to the if statement directly before the warning. If we add other outputs to this later we'll have to condense this anyway because it asks about a specific warning and this function won't necessarily always deal with this one warning.
```bash
if (!$this->getConfig()->get('hide_git_mode_warning') && ($mode === 'git')) {
    $this->log()->warning(
```

Reply:
Sure. I prefer the exit-early paradigm, but the variants you suggest are also clear and have merits as described. I'll alter as proposed.